src/proof_system/prover.rs

```diff
@@ -197,6 +197,17 @@ impl Prover {
         // the Transcript has been seeded with the preprocessed commitments
         let mut transcript = self.preprocessed_transcript.clone();
 
+        // PIs have to be part of the transcript
+        for pi in self.cs.to_dense_public_inputs().iter() {
+            transcript.append_scalar(b"pi", pi);
+        }
+
+        // We fill with zeros up to the domain size, just like the verifier will
+        // do
```

Comment:
I find this comment confusing since it is not immediately clear to me where the verifier will fill the `transcript` with zeros